illumos-utils/src/opte/non_illumos.rs

```diff
@@ -43,3 +70,268 @@ pub fn delete_all_xde_devices(log: &Logger) -> Result<(), Error> {
     slog::warn!(log, "`xde` driver is a fiction on non-illumos systems");
     Ok(())
 }
+
+#[derive(Debug)]
+pub(crate) struct RouteInfo {
+    pub dest: IpCidr,
+    pub target: RouterTarget,
+    pub class: RouterClass,
+}
+
+impl PartialEq for RouteInfo {
+    fn eq(&self, other: &Self) -> bool {
+        if self.dest != other.dest {
+            return false;
+        }
+        match (self.class, other.class) {
+            (RouterClass::System, RouterClass::Custom) => return false,
+            (RouterClass::Custom, RouterClass::System) => return false,
+            (_, _) => {}
+        }
+        match (self.target, other.target) {
+            (RouterTarget::Drop, RouterTarget::Drop) => true,
+            (
+                RouterTarget::InternetGateway(id0),
+                RouterTarget::InternetGateway(id1),
+            ) => id0 == id1,
+            (RouterTarget::Ip(ip0), RouterTarget::Ip(ip1)) => ip0 == ip1,
+            (
+                RouterTarget::VpcSubnet(cidr0),
+                RouterTarget::VpcSubnet(cidr1),
+            ) => cidr0 == cidr1,
+            (_, _) => false,
+        }
+    }
+}
+
+impl RouteInfo {
+    #[allow(dead_code)]
+    pub fn is_system_default_ipv4_route(&self) -> bool {
+        if self.dest
+            != IpCidr::Ip4(Ipv4Cidr::new(
+                Ipv4Addr::ANY_ADDR,
+                0.try_into().unwrap(),
+            ))
+        {
+            return false;
+        }
+        if !matches!(self.target, RouterTarget::InternetGateway(None)) {
+            return false;
+        }
+        matches!(self.class, RouterClass::System)
```

Comment:
This might be clearer using the `PartialEq` implementation above, IIUC. Something like

```rust
let system_default_route = RouteInfo {
    dest: IpCidr::Ip4(Ipv4Cidr::new(
                Ipv4Addr::ANY_ADDR,
                0.try_into().unwrap(),
            )),
    target: RouterTarget::InternetGateway(None),
    class: RouterClass::System,
};
*self == system_default_route
```

This would also be a little more future-proof against new fields showing up in `RouteInfo`.

Reply:
Done in 2024b245c